### PR TITLE
Parse supplement with date

### LINKED ISSUE #104: Parse supplement with date
NBS CIRC 24supJan1924
NBS CIRC 24supJuly1922
NBS HB 67suppFeb1965

_Originally posted by @mico in https://github.com/metanorma/nist-pubid/issues/94#issuecomment-1017157085_

### PR DESCRIPTION
**Pull request by mico**
closes #104 
related #125

### CODE REVIEW

File: spec/nist_pubid/document_spec.rb [hunk 1]
```diff
@@ -919,6 +919,20 @@
...
+    context "NBS CIRC 24supJan1924" do
+      let(:original_pubid) { "NBS CIRC 24supJan1924" }
+      let(:short_pubid) { "NBS CIRC 24supJan1924" }
+
+      it_behaves_like "converts pubid to different formats"
+    end
+
+    context "NBS HB 67suppFeb1965" do
+      let(:original_pubid) { "NBS HB 67suppFeb1965" }
+      let(:short_pubid) { "NBS HB 67supFeb1965" }
```
Comment (ronaldtse) [inline spec/nist_pubid/document_spec.rb:931]: I suppose we need to have a way to denote the publication date of a supplement. Can we do "NBS HB 67supe196502" to indicate it is edition "196502"?

Screenshot 2022-01-28 at 3 37 30 PM

Comment (mico) [reply spec/nist_pubid/document_spec.rb]: @ronaldtse we already represent edition month with 3 letters, like: "Feb1965".
example here: https://github.com/metanorma/nist-pubid/blob/main/spec/nist_pubid/document_spec.rb#L706
Should we change it to YYYYMM with only numbers?

Comment (ronaldtse) [reply spec/nist_pubid/document_spec.rb]: Yes, let's change to YYYYMM.

Comment (mico) [reply spec/nist_pubid/document_spec.rb]: Done

Comment (mico) [reply spec/nist_pubid/document_spec.rb]: > Yes, let's change to YYYYMM.

Done

**Pull request merged**